src/snowflake/cli/plugins/snowpark/package_utils.py

```diff
@@ -68,165 +66,183 @@ def deduplicate_and_sort_reqs(
-) -> List[RequirementWithFilesAndDeps]:
+def _check_for_shared_libraries(
+    dependencies: List[RequirementWithWheelAndDeps],
+) -> List[str]:
     return [
-        dep
-        for dep in dependencies
-        if dep.requirement.name not in [package.name for package in avaiable_in_conda]
+        dependency.requirement.name
+        for dependency in dependencies
+        if any(file.endswith(".so") for file in dependency.namelist())
     ]
 
 
-def generate_deploy_stage_name(identifier: str) -> str:
-    return (
-        identifier.replace("()", "")
-        .replace(
-            "(",
-            "_",
-        )
-        .replace(
-            ")",
-            "",
-        )
-        .replace(
-            " ",
-            "_",
-        )
-        .replace(
-            ",",
-            "",
-        )
-    )
-
-
-def _perform_native_libraries_check(deps: List[RequirementWithFilesAndDeps]):
-    if native_libraries := _check_for_native_libraries(deps):
-        _log_native_libraries(native_libraries)
+def _perform_shared_libraries_check(deps: List[RequirementWithWheelAndDeps]):
+    if native_libraries := _check_for_shared_libraries(deps):
+        _log_shared_libraries(native_libraries)
         return True
     else:
         log.info("Unsupported native libraries not found in packages (Good news!)...")
         return False
 
 
-def _log_native_libraries(
-    native_libraries: List[str],
+def _log_shared_libraries(
+    shared_libraries: List[str],
 ) -> None:
     log.error(
-        "Following dependencies utilise native libraries, not supported by Conda:"
+        "Following dependencies utilise shared libraries, not supported by Conda:"
     )
-    log.error("\n".join(set(native_libraries)))
-    log.error("You may still try to create your package, but it probably won`t work")
+    log.error("\n".join(set(shared_libraries)))
+    # TODO: add "with --allow-shared-libraries" flag when refactoring snowpark build command
+    log.error("You may still try to create your package, but it probably won't work")
     log.error("You may also request adding the package to Snowflake Conda channel")
     log.error("at https://support.anaconda.com/")
 
 
-def _confirm_native_libraries(allow_native_libraries: PypiOption) -> bool:
-    if allow_native_libraries == PypiOption.ASK:
-        return click.confirm("Continue with package installation?", default=False)
+def _log_dependencies_found_in_conda(available_dependencies: List[Requirement]) -> None:
+    if len(available_dependencies) > 0:
+        log.info(
+            "Good news! The following package dependencies can be"
+            " imported directly from Anaconda, and will be excluded from"
+            " the zip: %s",
+            format(", ".join(dep.name for dep in available_dependencies)),
+        )
     else:
-        return allow_native_libraries == PypiOption.YES
+        log.info("None of the package dependencies were found on Anaconda")
```

Comment:
I think the `package create` is a good example of command where we can utilse `phase` and `step` for showing intermediate output instead of using logging.

Reply:
I'll address that in PR refactoring `snow snowpark build`

Reply:
To quickly answer here: The problem with converting to "phase"/"step" is that we'll start printing on output without `--verbose` flag, or stop logging to avoid double messages on the output. I think it should be a part of separate refactor (replace `log.info` with `console.step/phase`)

Reply:
https://snowflakecomputing.atlassian.net/browse/SNOW-1271652